README.md

```diff
@@ -2,18 +2,198 @@
+
+#### Common Functions
+*The following commands should be run from the `layer0` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Run Unit tests       | `make unittest` |
+| Run Smoke Tests      | `make smoketest` |
+| Run System Tests     | `make systemtest` |
+
+
+# Tools 
+The following tools are used to help make development with Layer0 easier.
+
+## Gomock
+The [Gomock](https://github.com/golang/mock) tool is a mocking framework for Golang. 
+It uses code generation to create mockable objects. 
+If any changes are made to mocked interfaces, the code generation will need to run again.
+
+#### Common Functions
+*The following commands should be run from the `layer0/scripts` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Install Deps               | `make -f Makefile.mocks deps` |
+| Recreate all mocks         | `make -f Makefile.mocks all` |
+| Recreate a subset of mocks | `make -f Makefile.mocks <subset>` |
+
+## Go-decorator
+The [Go-Decorator](https://github.com/imshealth/go-decorator) tool implements the decorator pattern for Golang.
+It uses code generation to create decorated objects. 
+If any changes are made to decorated interfaces, the code generation will need to run again.
+
+#### Common Functions
+*The following commands should be run from the `layer0/scripts` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Install Deps                    | `make -f Makefile.decorators deps` |
+| Recreate all decorators         | `make -f Makefile.decorators all` |
+| Recreate a subset of decorators | `make -f Makefile.decorators <subset>` |
+
+## Flow
+Flow is a bash script that automates common workflows when developing with Layer0. 
+This tool requires that the environment variable `LAYER0_PREFIX` is set to the name of you Layer0 instance, 
+or that the `-p <prefix>` flag is used with any command.
+
+#### Common Functions
+*The following commands should be run from the `layer0/scripts` directory*
+
+| Function      | Command       |
+| ------------- |:--------------|
+| Build the API Docker image, push it to Dockerhub, and run the new image as your Layer0 API | `flow.sh api` |
+| Build the Runner Docker image, push it to Dockerhub, and run the new image as your Layer0 Runner | `flow.sh runner`|
+| Delete all entities in your Layer0 that aren't the API | `flow.sh delete` |
+| Run all jobs that are in the `PENDING` state (only for use with local development) | `flow.sh runjobs` |
+
+# Development Workflow
+A common workflow for working with Layer0
+
+####  Install a Layer0 Instance
+This is only required if you don't already have a Layer0 instance.
+We will use the Layer0 instances to test our changes AWS.
+
+```
+# install a new Layer0
+$ go run setup/main.go apply <instance>
```

Comment:
Does this work? If run from `/layer0` directory, it appears that the apply action looks for templates in `/layer0` directory and faies. Should we direct developers to run from `/layer0/setup` directory instead?